pkg/sriov/sriov.go

```diff
@@ -244,14 +205,27 @@ func (s *sriovManager) ApplyVFConfig(conf *sriovtypes.NetConf) error {
 	}
 
 	// 1. Set vlan
-	if conf.Vlan != 0 || conf.VlanQoS != 0 {
-		// ensure that VLAN QoS is in the 0-7 range
-		if conf.VlanQoS < 0 || conf.VlanQoS > 7 {
-			return fmt.Errorf("failed to set vf %d vlan QoS field %d: QoS PCP value must be in the range 0-7", conf.VFID, conf.VlanQoS)
-		}
+	// validate config
+	if conf.Vlan == 0 && conf.VlanQoS != 0 {
+		return fmt.Errorf("non-zero vlan id must be configured to set vlan QoS")
+	}
 
-		if err = s.nLink.LinkSetVfVlanQos(pfLink, conf.VFID, conf.Vlan, conf.VlanQoS); err != nil {
-			return fmt.Errorf("failed to set vf %d vlan configuration: %v", conf.VFID, err)
+	if conf.Vlan != 0 {
+		// set vlan qos if present in the config
+		if conf.VlanQoS != 0 {
+			// validate that VLAN QoS is in the 0-7 range
+			if conf.VlanQoS < 0 || conf.VlanQoS > 7 {
```

Comment:
I'd prefer to have this validation check in the config.LoadConf() and throw errors earlier than here. Once we get to  ApplyVFConfig() method we could safely assume that configs are valid and apply vlanqos if it's `Vlan` and `VlanQoS` both are present.
Just looking into  [ApplyVFConfig ](https://github.com/intel/sriov-cni/blob/01ab257c45960c6c6b1d4b9501bf7991fe16e9d7/pkg/config/config.go#L20) method and I see there's an opportunity to validate a vlan ids as well there if this is supplied.

Reply:
Good idea! I'll move it there.

Do you want me to include that VLAN ID range validation (1-4094) in this PR as well? Or leave it for now?

Reply:
> Good idea! I'll move it there.
> 
> Do you want me to include that VLAN ID range validation (1-4094) in this PR as well? Or leave it for now?

That would be great. There are some other constraints on valid range though according to [cisco hw](https://www.orbit-computer-solutions.com/vlan-id-ranges-explained/) on a network. Please see [here]. But I don't think we need to go at that length right now. For now we can just validate against the above range.

Reply:
Cisco VLAN constraints come from their proprietary ISL protocol. I'll go with the 0-4094 range then (1-4094 as valid range and 0 as default/disabled).